# B1: Credit Scoring and PD Estimation
**Track:** Resume B - Quantitative Risk & Analytics
**Environment:** resume-b
**Started:** 2025-05
**Build Guide Reference:** AllProjectsBuildGuide.md — Section B1

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import matplotlib
import warnings
import os

matplotlib.rcParams['figure.dpi'] = 150
matplotlib.rcParams['figure.figsize'] = (10, 5)
matplotlib.rcParams['axes.spines.top'] = False
matplotlib.rcParams['axes.spines.right'] = False
pd.set_option('display.max_columns', 50)
pd.set_option('display.float_format', '{:.4f}'.format)
warnings.filterwarnings('ignore')

DATA_DIR = '../data/'
OUTPUT_DIR = '../outputs/'
os.makedirs(OUTPUT_DIR, exist_ok=True)

def savefig(filename, tight=True):
    path = os.path.join(OUTPUT_DIR, filename)
    plt.savefig(path, bbox_inches='tight' if tight else None, dpi=150)
    print(f"Saved: {filename}")

print("Environment ready.")

Environment ready.


In [2]:
df = pd.read_csv(os.path.join(DATA_DIR, 'application_train.csv'))

print("Shape:", df.shape)
print()

default_rate = df['TARGET'].mean()
class_counts = df['TARGET'].value_counts()

print("Class distribution:")
print(f"  Non-default (0): {class_counts[0]:,}  ({1 - default_rate:.1%})")
print(f"  Default     (1): {class_counts[1]:,}  ({default_rate:.1%})")
print()
print(f"Majority-class baseline accuracy: {max(default_rate, 1 - default_rate):.2%}")
print()
print("Column count by dtype:")
print(df.dtypes.value_counts())
print()
print("Missing value summary (top 15 columns by % missing):")
missing = (df.isnull().sum() / len(df) * 100).sort_values(ascending=False)
print(missing[missing > 0].head(15).round(2).to_string())

Shape: (307511, 122)

Class distribution:
  Non-default (0): 282,686  (91.9%)
  Default     (1): 24,825  (8.1%)

Majority-class baseline accuracy: 91.93%

Column count by dtype:
float64    65
int64      41
str        16
Name: count, dtype: int64

Missing value summary (top 15 columns by % missing):
COMMONAREA_AVG             69.8700
COMMONAREA_MODE            69.8700
COMMONAREA_MEDI            69.8700
NONLIVINGAPARTMENTS_MEDI   69.4300
NONLIVINGAPARTMENTS_MODE   69.4300
NONLIVINGAPARTMENTS_AVG    69.4300
FONDKAPREMONT_MODE         68.3900
LIVINGAPARTMENTS_AVG       68.3500
LIVINGAPARTMENTS_MEDI      68.3500
LIVINGAPARTMENTS_MODE      68.3500
FLOORSMIN_MODE             67.8500
FLOORSMIN_AVG              67.8500
FLOORSMIN_MEDI             67.8500
YEARS_BUILD_AVG            66.5000
YEARS_BUILD_MODE           66.5000


## Data Sources

| File | Source URL | Downloaded | Coverage |
|---|---|---|---|
| application_train.csv | https://www.kaggle.com/c/home-credit-default-risk | 2025-05-14 | 307,511 loan applications, 122 features, 8.1% default rate |

**Majority-class baseline accuracy: 91.93%**
All model performance claims are stated as lift over this baseline.